lib/kubernetes-deploy/resource_cache.rb

```diff
@@ -0,0 +1,64 @@
+# frozen_string_literal: true
+
+require 'concurrent/hash'
+
+module KubernetesDeploy
+  class ResourceCache
+    def initialize(namespace, context, logger)
+      @namespace = namespace
+      @context = context
+      @logger = logger
+
+      @kind_fetcher_locks = Concurrent::Hash.new { |hash, key| hash[key] = Mutex.new }
```

Comment:
Hmmm. I'm not sure this is now fine. There were two race conditions before: Two threads accessing different keys of the hash in parallel. And the other one was about two threads racing to evaluate the "cache miss" block here. I think the `Concurrent::Hash` only protects against the former, but not the latter.

Reply:
I'm not sure for non-MRI, and I'm not sure how to find out either. For MRI I think the GIL should still have us covered, and earlier today I ran my only-makes-one-call test with 500 resources of 3 different kind over 5,000 times in a row and it passed every time. That doesn't 100% prove anything, but it seems more than good enough for sure.